src/mbgl/map/tile_parser.cpp

```diff
@@ -50,13 +50,13 @@ TileParser::TileParser(const std::string &data, VectorTileData &tile_,
     assert(collision);
 }
 
-void TileParser::parse() {
-    parseStyleLayers(style->layers);
+void TileParser::parse(float z) {
+    parseStyleLayers(style->layers, z);
```

Comment:
I _think_ you can avoid needing to thread `z` through by using `tile.id.z` instead.